### PR TITLE
crowbar-pacemaker: Allow overriding start/stop commands for crm_resource

### PR DESCRIPTION
**Pull request by rhafer**
The 1st commit is just a pretty simple fix for `service_is_running?` to make it work with Master/Slave resources.

The 2nd commit enhances the pacemaker "service" provider to allow passing in alternative start/stop commands when used with the "restart_crm_resource" flag. We need this for the galera resource agent. In order to restart mysql on a single node we need to get the resource though a demote/promote cycle once (instead of start/stop). See e.g. https://github.com/jhesketh/crowbar-openstack/commit/ce00239fe10848b591a561a16d28a43d7f68a5a7 for the current work in progress.

### CODE REVIEW

**Comment by rhafer** [comment]
[P11 galera config: Find the best value for SST](https://trello.com/c/27Q5XW7K/28-p11-galera-config-find-the-best-value-for-sst)

File: chef/cookbooks/crowbar-pacemaker/providers/service.rb [hunk 1]
```diff
@@ -221,10 +221,14 @@
 def service_is_running?(name, use_crm_resource, pacemaker_resource)
   if use_crm_resource
     `crm_resource --force-check --resource #{pacemaker_resource}`
+    # For Master/Slave resources "monitor" will return OCF_RUNNING_MASTER (8)
+    # on nodes that are running the resource in Master role currently. We need
+    # to treat that as a successfully result as well.
+    $?.success? || $?.exitstatus == 8
```
Comment (nicolasbock) [inline chef/cookbooks/crowbar-pacemaker/providers/service.rb]: Wow, that is certainly unusual :frowning_face:.

Comment (jsuchome) [inline chef/cookbooks/crowbar-pacemaker/providers/service.rb]: Don't we need to limit this to certain services only?

Comment (rhafer) [reply chef/cookbooks/crowbar-pacemaker/providers/service.rb]: Why? OCF_RUNNING_MASTER is only ever returned for Master/Slave resources.

**Comment by jhesketh** [comment]
This looks good to me.

File: chef/cookbooks/crowbar-pacemaker/providers/service.rb (hunk 1 above)
Comment (jsuchome) [reply chef/cookbooks/crowbar-pacemaker/providers/service.rb]: OK.

**Pull request merged**